userguide/cpp.md

```diff
@@ -0,0 +1,198 @@
+* `-std=c++11` - some headers exposed by Turi Create require the use of C++11 or later.
+* `-stdlib=libc++` - may be needed on macOS (probably not on Linux).
+* Assuming PATH_TO_TC refers to the location where you cloned the Turi Create repository, the following directories must be added to the include paths with `-I`, and to the link paths with `-L`:
+  * `-IPATH_TO_TC/turicreate/src`
+  * `-IPATH_TO_TC/turicreate/deps/local/include`
+  * `-IPATH_TO_TC/turicreate/src/external/armadillo/include`
+  * `-LPATH_TO_TC/turicreate/release/src/unity`
+* Depending on your compiler toolchain, the following defines may be needed (note: these are autodetected for Turi Create itself with `./configure`, but the values must be passed on to your own build process:
+* `-DHASH_FOR_UINT128_DEFINED` - set this if your compiler toolchain supports `unsigned __int128`.
+* `-DHASH_FOR_INT128_DEFINED` - set this if your compiler toolchain supports `__int128`.
+* `-lunity_shared` to link against `libunity_shared.so`.
+* `PATH_TO_TC/turicreate/release/src/unity/libunity.a` to statically link libunity.a into your project. Note that this is needed in addition to the dynamic library listed above.
+
+##### Example: make predictions in C++ from a model trained in Python
+
+Let's assume we're starting with the model that results from the introductory example in the [boosted trees classifier](https://github.com/apple/turicreate/blob/3490286b27ff5d79cb90d09fe026d5671ce990c7/userguide/supervised-learning/boosted_trees_classifier.md#gradient-boosted-regression-trees) section of the user guide. The model is available as [mushroom.tcmodel.zip](https://github.com/apple/turicreate/files/1654991/mushroom.tcmodel.zip) as well.
+
+To make predictions from this model in Python, you could run:
+```python
+import turicreate as tc
+data =  tc.SFrame.read_csv('turicreate/lfs/datasets/xgboost/mushroom.csv')
+model = tc.load_model('temp4/mushroom.tcmodel')
+data[0]
+```
+```
+{'bruises?': 't',
+ 'cap-color': 'n',
+ 'cap-shape': 'x',
+ 'cap-surface': 's',
+ 'gill-attachment': 'f',
+ 'gill-color': 'k',
+ 'gill-size': 'n',
+ 'gill-spacing': 'c',
+ 'habitat': 'u',
+ 'label': 'c',
+ 'odor': 'p',
+ 'population': 's',
+ 'ring-number': 'o',
+ 'ring-type': 'p',
+ 'spore-print-color': 'k',
+ 'stalk-color-above-ring': 'w',
+ 'stalk-color-below-ring': 'w',
+ 'stalk-root': 'e',
+ 'stalk-shape': 'e',
+ 'stalk-surface-above-ring': 's',
+ 'stalk-surface-below-ring': 's',
+ 'veil-color': 'w',
+ 'veil-type': 'p'}
+```
+```python
+model.predict_topk(data[0], k=2, output_type='probability')
+```
+```
+Columns:
+	id	int
+	class	int
+	probability	float
+
+Rows: 2
+
+Data:
++----+-------+---------------+
+| id | class |  probability  |
++----+-------+---------------+
+| 0  |   1   | 0.57558375597 |
+| 0  |   0   | 0.42441624403 |
++----+-------+---------------+
+[2 rows x 3 columns]
+```
+
+In C++, the equivalent code would be (minus the CSV parsing):
+```cpp
+// Assuming this is in a sibling directory of turicreate,
```

Comment:
It seems a little strange to me that you have shell commands as cpp comments. Perhaps you should just add a shell section at the end of this section and move these shell commands there.